parlai/core/torch_agent.py

```diff
@@ -606,7 +611,7 @@ def __init__(self, opt: Opt, shared=None):
             if not shared and opt['gpu'] != -1:
                 torch.cuda.set_device(opt['gpu'])
         # indicate whether using fp16
-        self.fp16 = self.use_cuda and self.opt.get('fp16', False)
```

Comment:
This is where it was dying before?

Reply:
Well this would be set as true, and then eventually it would call `parlai.utils.torch.fp16_optimizer` and that will throw an import error saying the user needs to install APEX.